src/io/flutter/view/BoolServiceExtensionCheckbox.java

```diff
@@ -19,45 +16,36 @@
  * enabling embedding UI to turn on and off the service extension
  * in a JPanel.
  */
-public class BoolServiceExtensionCheckbox implements Disposable {
+public class BoolServiceExtensionCheckbox {
 
-  private final EventStream<Boolean> currentValue;
   private final JCheckBox checkbox;
-  private StreamSubscription<Boolean> currentValueSubscription;
 
   BoolServiceExtensionCheckbox(FlutterApp app, @NotNull String extensionCommand, String label, String tooltip) {
     checkbox = new JCheckBox(label);
     checkbox.setHorizontalAlignment(JLabel.LEFT);
     checkbox.setToolTipText(tooltip);
     assert(app.getVMServiceManager() != null);
-    currentValue = app.getVMServiceManager().getServiceExtensionState(extensionCommand);
-    app.hasServiceExtension(extensionCommand, checkbox::setEnabled, this);
+    app.hasServiceExtension(extensionCommand, checkbox::setEnabled);
 
     checkbox.addActionListener((l) -> {
       final boolean newValue = checkbox.isSelected();
-      if (currentValue.setValue(newValue)) {
-        if (app.isSessionActive()) {
-          app.callBooleanExtension(extensionCommand, newValue);
-        }
+      app.getVMServiceManager().setServiceExtensionState(
+        extensionCommand,
+        newValue,
+        newValue);
+      if (app.isSessionActive()) {
+        app.callBooleanExtension(extensionCommand, newValue);
       }
     });
 
-    currentValueSubscription = currentValue.listen((value) -> {
-      if (checkbox.isSelected() != value) {
-        checkbox.setSelected(value);
+    app.getVMServiceManager().getServiceExtensionState(extensionCommand).listen((state) -> {
+      if (checkbox.isSelected() != state.isEnabled()) {
+        checkbox.setSelected(state.isEnabled());
       }
     }, true);
   }
 
   JCheckBox getComponent() {
     return checkbox;
   }
-
```

Comment:
add back this dispose method. you caught a big that we aren't calling this dispose method not that it isn't needed. otherwise we are leaking memory